src/NUnitFramework/framework/Internal/Execution/CompositeWorkItem.cs

```diff
@@ -231,13 +240,50 @@ private void RunChildren()
 
         private void CreateChildWorkItems()
         {
-            _children = new List<WorkItem>();
+            Children = new List<WorkItem>();
 
             foreach (ITest test in _suite.Tests)
-                if (_childFilter.Pass(test))
-                    _children.Add(WorkItem.CreateWorkItem(test, _childFilter));
+                if (_childFilter.Pass(test) && test.Properties.ContainsKey(PropertyNames.Order))
+                {
+                        Children.Add(WorkItem.CreateWorkItem(test, _childFilter));
+                        _countOrder++;
+                }
+
+            foreach (ITest test in _suite.Tests)
+                if (_childFilter.Pass(test) && !test.Properties.ContainsKey(PropertyNames.Order))
+                {
+                    Children.Add(WorkItem.CreateWorkItem(test, _childFilter));
+                }
```

Comment:
Why two separate loops? List.Insert will allow you to stick things at the front of the list as you go through the tests once.